src/gsi/pcgsoi.f90

```diff
@@ -161,6 +162,11 @@ subroutine pcgsoi()
   use berror, only: vprecond
 
   use stpjomod, only: stpjo_setup
+  
+  use gridmod,only: l_reg_update_hydro_delz
+  use guess_grids, only:geom_hgti,geom_hgti_bg
```

Comment:
While compiling in debug mode, variables geom_hgti and geom_hgti_bg were identified as unused.  Please remove unused variables.

Reply:
deleted two variables  geom_hgti and geom_hgti_bg

Reply:
Somehow it is the change to use variable l_reg_update_hydro_delz is not visible anywhere? Is this really needed at the l_reg_update_hydro_delz level? Looks odd.

Reply:
l_reg_update_hydro_delz is a new namelist parameter. It can be found in gsimod.f90.

Reply:
I will do the test to move this part to read_guess. Thanks for your suggestion.

Reply:
finish the code change to move the l_reg_update_hydro_delz related code to update_guess.f90 from pcgsoi.f90.